lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -224,11 +224,28 @@ LogicalResult AndOp::canonicalize(AndOp op, PatternRewriter &rewriter) {
   auto size = inputs.size();
   assert(size > 1 && "expected 2 or more operands, `fold` should handle this");
 
-  // TODO: remove all duplicate arguments
-  // and(..., x, x) -> and(..., x) -- idempotent
-  if (inputs[size - 1] == inputs[size - 2]) {
-    rewriter.replaceOpWithNewOp<AndOp>(op, op.getType(), inputs.drop_back());
-    return success();
+  // and(..., x, ..., x) -> and(..., x, ...) -- idempotent
+  switch (inputs.size()) {
+    case 1:
+    case 2:
+      // Trivial and(x), and(x, x) cases are handled by [AndOp::fold] above.
+      break;
+    default: {
```

Comment:
Please simplify this into an if, just use `if (inputs.size() > 2) {`, otherwise, LGTM, thanks!

Reply:
Done.

Meta question: In general, is it preferable to address small comments as such with a force-push (to keep commit histories clean), or just an additional patch on top?

Reply:
We do "squash and merge" so an additional path is always ok!

Thank you for the contributions and welcome to the community.  If you'd like to get commit access, feel free to request (check out the "LLVM Developer Policy" doc) thanks!